File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/teacher/model/TeacherCreateRequestDTO.java
package kh.farrukh.sortmapperexamplejava.teacher.model;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;

public record TeacherCreateRequestDTO(
    @NotBlank
    @Size(min = 2, max = 50)
    String firstName,
    @NotBlank
    @Size(min = 2, max = 50)
    String lastName,
    @Size(min = 2, max = 50)
    String middleName,
    @Positive
    Integer age
) {

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/teacher/model/TeacherCreateRequestDTO.kt
package kh.farrukh.sortmapperexample.teacher.model

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.Size

data class TeacherCreateRequestDTO(
    @field:[NotBlank Size(min = 2, max = 50)]
    val firstName: String,
    @field:[NotBlank Size(min = 2, max = 50)]
    val lastName: String,
    @field:Size(min = 2, max = 50)
    val middleName: String?,
    @field:[NotNull Positive]
    val age: Int
)
